cmd/api/main.go

```diff
@@ -49,9 +54,32 @@ func run() error {
 	//create and init http server:
 	server := bookhttp.NewServer(bookhttp.ServerConfig{Port: 8080}, bookHandler)
 
-	err = server.ListenAndServe()
-	if err != nil && !errors.Is(err, http.ErrServerClosed) {
-		return fmt.Errorf("unexpected http server error: %w", err)
+	go func() {
+		err := server.ListenAndServe()
+		if err != nil && !errors.Is(err, http.ErrServerClosed) {
+			log.Printf("unexpected http server error: %v", err)
+		}
+		log.Println("stopped serving new requests.")
+	}()
+
+	sc := make(chan os.Signal, 1)
+	signal.Notify(sc, syscall.SIGINT, syscall.SIGTERM)
+	<-sc
+
+	deadline := 10
+	deadlineStr := os.Getenv("DEADLINE")
```

Comment:
Lets go with `SERVICE_SHUTDOWN_TIMEOUT`.

Reply:
Lets use the `time.Duration` type. You can use this function to parse it https://pkg.go.dev/time#ParseDuration .